Sources/Puree/Output/BufferedOutput.swift

```diff
@@ -160,8 +173,25 @@ open class BufferedOutput: InstantiatableOutput {
         let newBuffer = Set(buffer.dropFirst(logCount))
         let dropped = buffer.subtracting(newBuffer)
         buffer = newBuffer
-        let chunk = Chunk(logs: dropped)
-        callWriteChunk(chunk)
+        let logsToSend: Set<LogEntry>
+        if let chunkDataSizeLimit = configuration.chunkDataSizeLimit {
+            var logsUnderSizeLimit = Set<LogEntry>()
+
+            var currentTotalLogSize = 0
+            for log in dropped {
+                if currentTotalLogSize + (log.userData?.count ?? 0) < chunkDataSizeLimit {
+                    logsUnderSizeLimit.insert(log)
+                    currentTotalLogSize += log.userData?.count ?? 0
+                } else {
+                    buffer = dropped.subtracting(logsUnderSizeLimit)
```

Comment:
What will cause when the situation if a single log reaches the limit?
I concern about the logs are remained in log stores forever.